src/material/datepicker/date-range-input.ts

```diff
@@ -0,0 +1,231 @@
+import {FocusOrigin} from '@angular/cdk/a11y';
+
+let nextUniqueId = 0;
+
+// Boilerplate for applying mixins to MatDateRangeInput.
+/** @docs-private */
+class MatDateRangeInputBase {
+  constructor(public _defaultErrorStateMatcher: ErrorStateMatcher,
+              public _parentForm: NgForm,
+              public _parentFormGroup: FormGroupDirective,
+              /** @docs-private */
+              public ngControl: NgControl) {}
+}
+const _MatDateRangeInputMixinBase: CanDisableCtor &
+    CanUpdateErrorStateCtor & typeof MatDateRangeInputBase =
+    mixinErrorState(mixinDisabled(MatDateRangeInputBase));
+
+@Component({
+  selector: 'mat-date-range-input',
+  templateUrl: 'date-range-input.html',
+  styleUrls: ['date-range-input.css'],
+  exportAs: 'matDateRangeInput',
+  inputs: ['disabled'],
+  host: {
+    'class': 'mat-date-range-input'
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {provide: MatFormFieldControl, useExisting: MatDateRangeInput},
+    {provide: NG_VALUE_ACCESSOR, useExisting: MatDateRangeInput, multi: true},
+    {provide: NG_VALIDATORS, useExisting: MatDateRangeInput, multi: true}
+  ]
+})
+export class MatDateRangeInput<D>  extends _MatDateRangeInputMixinBase implements
+  MatFormFieldControl<DateRange<D>>, ControlValueAccessor, Validator, CanUpdateErrorState,
+  CanDisable, OnDestroy {
+
+  /** Current value of the range input. */
+  value: DateRange<D> | null = null;
+
+  /** Emits when the input's state has changed. */
+  stateChanges = new Subject<void>();
+
+  /** Unique ID for the input. */
+  id = `mat-date-range-input-${nextUniqueId++}`;
+
+  /** Whether the control is focused. */
+  focused = false;
+
+  /** Whether the control's label should float. */
+  shouldLabelFloat = true;
+
+  /** Name of the form control. */
+  controlType = 'mat-date-range-input';
+
+  /**
+   * Implemented as a part of `MatFormFieldControl`, but not used.
+   * Use `startPlaceholder` and `endPlaceholder` instead.
+   * @docs-private
+   */
+  placeholder: string;
+
+  /** Whether the input is required. */
+  @Input()
+  get required(): boolean { return !!this._required; }
+  set required(value: boolean) {
+    this._required = coerceBooleanProperty(value);
+  }
+  private _required: boolean;
+
+  /** Whether the datepicker input is empty. */
+  get empty(): boolean {
+    // TODO(crisbeto): should probably look at the input values as well.
+    return !this._model || !this._model.selection;
+  }
+
+  /** Value for the `aria-describedby` attribute of the inputs. */
+  _ariaDescribedBy: string | null = null;
+
+  /** Value for the `aria-labelledby` attribute of the inputs. */
+  _ariaLabelledBy: string | null = null;
+
+  /** Placeholder for the start input. */
+  @Input() startPlaceholder: string;
+
+  /** Placeholder for the end input. */
+  @Input() endPlaceholder: string;
+
+  /** Separator text to be shown between the inputs. */
+  @Input() separator = '—';
```

Comment:
```suggestion
  @Input() separator = '–';
```
Should be `'–'` (`U+2013`) instead of `'—'` (`U+2014`) to follow proper formal English writing conventions.

Reply:
I agree that an en-dash would look more normal here and should probably be the default

Reply:
I don't have a preference here. I went with this one, because it's what we had on the paginator.